src/Settings.cpp

```diff
@@ -142,6 +148,24 @@ ADDON_STATUS Settings::ReadBackendSettings()
   return ADDON_STATUS_OK;
 }
 
+void Settings::SetConnection(bool status)
+{
+  if (status == true)
+  {
+      kodi::vfs::CFile outputFile;
+      if (outputFile.OpenFileForWrite(connectionFlag))
+      {
+        outputFile.Close();
+      }
+      m_connectionConfirmed = true;
```

Comment:
About here and only as idea, it can also make a bit smaller (and without the extra 2 spaces ;) )
```cpp
  if (status == true)
  {
    kodi::vfs::CFile outputFile;
    m_connectionConfirmed = outputFile.OpenFileForWrite(connectionFlag);
  }
```
The Close not direct needed as done by his destructor.

The Close can also be called, but more for cases as CFile is in class values or after read/write comes bigger work where takes time.

Reply:
Ok changed but I set the state to true even if the write fails because at this point it has connected to the backend.